src/metabase/query_processor/middleware/escape_join_aliases.clj

```diff
@@ -1,15 +1,214 @@
+  the original alias is identical to the escaped alias; that's not useful information to include in `:info`."
+  [query]
+  (let [escaped->original-maps (mbql.u/match query
+                                 (m :guard (every-pred map? ::original->escaped))
+                                 (merge
+                                  (set/map-invert (::original->escaped m))
+                                  (merged-escaped->original-with-no-ops-removed (dissoc m ::original->escaped))))]
+    (not-empty
+     (into {}
+           (comp cat
+                 (remove (fn [[k v]]
+                           (= k v))))
+           escaped->original-maps))))
+
+(defn- add-escaped->original-info
+  "Add a map of escaped alias -> original alias under `[:info :alias/escaped->original]`; this is used
+  by [[restore-aliases]] below."
+  [query]
+  (let [escaped->original (not-empty (merged-escaped->original-with-no-ops-removed query))]
+    (cond-> query
+      escaped->original (assoc-in [:info :alias/escaped->original] escaped->original))))
+
+(defn- replace-original-aliases-with-escaped-aliases
+  "'Commit' all the new escaped aliases we determined we should use to the query, and clean up all the keys we added in
+  the process of determining this information.
+
+  * Replace the `:join-alias` in `:field` clauses with the `::join-alias` and remove `::join-alias`.
+
+  * Replace the `:alias` in join clauses with the `::alias` and remove `::alias`.
+
+  * Remove the `::original->escaped` maps.
+
+  You might be asking, why don't we just do this in the first place rather than adding all these extra keys that we
+  eventually remove? For joins, we need to track the original alias for a while to build the `::original->escaped`
+  map. For `:field` clauses, we need to keep track of whether we already escaped it or not , since the mapping between
+  original alias and escaped alias might be different based on the level of query we're at."
+  [query]
+  (mbql.u/replace query
+    ;; update inner queries that have `::original->escaped` maps
+    (m :guard (every-pred map? ::original->escaped))
+    (-> (dissoc m ::original->escaped)
+        ;; recursively update source levels and `:field` clauses.
+        replace-original-aliases-with-escaped-aliases)
+
+    ;; update joins
+    (m :guard (every-pred map? ::alias))
+    (-> m
+        (assoc :alias (::alias m))
+        (dissoc ::alias)
+        ;; recursively update source levels and `:field` clauses.
+        replace-original-aliases-with-escaped-aliases)
+
+    ;; update `:field` clauses
+    [:field id-or-name (options :guard (every-pred map? ::join-alias))]
+    [:field id-or-name (-> options
+                           (assoc :join-alias (::join-alias options))
+                           (dissoc ::join-alias))]))
+
+(defn escape-join-aliases
+  "Pre-processing middleware. Make sure all join aliases are unique, regardless of case (some databases treat table
```

Comment:
Just by looking at the name I wouldn't expect this to generate unique aliases. It sounds like as if it just worried about special characters in names.